roqoqo/tests/integration/operations/pragma_operations.rs

```diff
@@ -3109,3 +3109,176 @@ fn pragma_change_device_substitute_trait() {
+    let pragma = PragmaControlledCircuit::new(1, Circuit::default());
+    let pragma_test = PragmaControlledCircuit::new(1, Circuit::default());
+
+    // (1) Substitute parameters function
+    let mut substitution_dict: Calculator = Calculator::new();
+    substitution_dict.set_variable("ro", 0.0);
+    let result = pragma_test
+        .substitute_parameters(&substitution_dict)
+        .unwrap();
+    assert_eq!(pragma, result);
+
+    // (2) Remap qubits function with an empty circuit
+    let mut qubit_mapping_test: HashMap<usize, usize> = HashMap::new();
+    qubit_mapping_test.insert(0, 2);
+    qubit_mapping_test.insert(2, 0);
+    let mut new_qubit_paulis: HashMap<usize, usize> = HashMap::new();
+    new_qubit_paulis.insert(2, 1);
+    let result = pragma_test.remap_qubits(&qubit_mapping_test).unwrap();
+    assert_eq!(result, pragma);
+
+    // (3) Remap qubits function with an non-empty circuit
+    let mut circuit = Circuit::new();
+    circuit.add_operation(PauliX::new(2));
+    let mut circuit_test = Circuit::new();
+    circuit_test.add_operation(PauliX::new(0));
+    let pragma = PragmaControlledCircuit::new(1, circuit_test);
+    let result = pragma.remap_qubits(&qubit_mapping_test).unwrap();
+    let test_gate = PragmaControlledCircuit::new(1, circuit);
+    assert_eq!(result, test_gate);
+}
+
+/// Test PragmaConditional Serialization and Deserialization traits (readable)
+#[cfg(feature = "serialize")]
+#[test]
+fn pragma_controlled_circuit_serde_readable() {
+    let pragma_serialization = PragmaControlledCircuit::new(1, Circuit::default());
+    assert_tokens(
+        &pragma_serialization.readable(),
+        &[
+            Token::Struct {
+                name: "PragmaControlledCircuit",
+                len: 2,
+            },
+            Token::Str("controlling_qubit"),
+            Token::U64(1),
+            Token::Str("circuit"),
+            Token::Struct {
+                name: "Circuit",
+                len: 3,
+            },
+            Token::Str("definitions"),
+            Token::Seq { len: Some(0) },
+            Token::SeqEnd,
+            Token::Str("operations"),
+            Token::Seq { len: Some(0) },
+            Token::SeqEnd,
+            Token::Str("_roqoqo_version"),
+            Token::Struct {
+                name: "RoqoqoVersionSerializable",
+                len: 2,
+            },
+            Token::Str("major_version"),
+            Token::U32(1),
+            Token::Str("minor_version"),
+            Token::U32(0),
+            Token::StructEnd,
+            Token::StructEnd,
+            Token::StructEnd,
+        ],
+    );
+}
+
+/// Test PragmaConditional Serialization and Deserialization traits (compact)
+#[cfg(feature = "serialize")]
+#[test]
+fn pragma_controlled_circuit_serde_compact() {
+    let pragma_serialization = PragmaControlledCircuit::new(1, Circuit::default());
+    assert_tokens(
+        &pragma_serialization.readable(),
```

Comment:
This one also says `readable()`, should be `compact()`?

Reply:
Fixed.